payments-core/src/main/java/com/stripe/android/googlepaylauncher/Tasks.kt

```diff
@@ -0,0 +1,31 @@
+package com.stripe.android.googlepaylauncher
+
+import com.google.android.gms.tasks.CancellationTokenSource
+import com.google.android.gms.tasks.Task
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.util.concurrent.Executor
+import kotlin.coroutines.resume
+
+internal suspend fun <T> Task<T>.awaitTask(cancellationTokenSource: CancellationTokenSource? = null): Task<T> {
+    return if (isComplete) {
+        this
+    } else {
+        suspendCancellableCoroutine { cont ->
+            // Run the callback directly to avoid unnecessarily scheduling on the main thread.
+            addOnCompleteListener(DirectExecutor, cont::resume)
+
+            cancellationTokenSource?.let { cancellationSource ->
+                cont.invokeOnCancellation { cancellationSource.cancel() }
+            }
+        }
+    }
+}
```

Comment:
Not necessary for this PR, but I added `kotlinx-coroutines-play-services` for `GooglePayRepository` in my refactor for that class. Should we align and use that instead?